miner/taiko_worker.go

```diff
@@ -335,7 +336,13 @@ loop:
 		}
 	}
 
-	return lastTransaction
+	if txsPruningResult == nil {
```

Comment:
If the `len(env.txs) == 101` and didn't touch line314 the final check is lost then the result is not what we want.

Reply:
Fixed in a new [PR](https://github.com/taikoxyz/taiko-geth/pull/408).